gc3apps/bf.uzh.ch/gcsvBulk.py

```diff
@@ -0,0 +1,216 @@
+    newly-created jobs so that this limit is never exceeded.
+    """
+
+    def __init__(self):
+        SessionBasedScript.__init__(
+            self,
+            version = __version__, # module version == script version
+            application = GcsvBulkApplication,
+            # only display stats for the top-level policy objects
+            # (which correspond to the processed files) omit counting
+            # actual applications because their number varies over
+            # time as checkpointing and re-submission takes place.
+            stats_only_for = GcsvBulkApplication,
+            )
+
+    def setup_args(self):
+        self.add_param('mfunction', help=(
+            "Name of the matlab function to run."
+            "A corresponding `.m` file must exist in the current directory."))
+        self.add_param('csvfile', help=(
+            "Name of the csv filte to be processed."))
+        self.add_param('N', default=0, type=int, help=("Value for the number of nodes."))
+
+    def get_function_name_and_file(self, mfunction):
+        if mfunction.endswith('.m'):
+            mfunction_m = mfunction
+            mfunction = mfunction[:-len('.m')]
+        else:
+            mfunction_m = mfunction + '.m'
+        m_file = join(os.getcwd(), mfunction_m)
+        if not exists(m_file):
+            raise ArgumentError(
+                self.actions['mfunction'],
+                ("Cannot read file '{m_file}'"
+                 " providing matlab function '{mfunction}'.")
+                .format(**locals()))
+        return mfunction, m_file
+
+    def get_csv_file(self, csvFile):
+        if not csvFile.endswith('.csv'):
+            csvFile += '.csv'
+        csv_file = join(os.getcwd(), csvFile)
+        if not exists(csv_file):
+            raise ArgumentError(
+                self.actions['csvfile'],
+                ("Cannot read file '{csv_file}.")
+                .format(**locals()))
+        return csv_file
+
+    def get_BlockSize(self, nrBlocks, csvFile):
+        with open(csvFile) as f:
+            for i, l in enumerate(f, 1):
+                pass
+        return ceil(i/nrBlocks)
+
+
+
+    def new_tasks(self, extra):
+        """
+        For each N, create an instance of GcsvBulkApplication
+        """
+        tasks = []
+
+        input_function, input_file = self.get_function_name_and_file(self.params.mfunction)
+        csv_file = self.get_csv_file(self.params.csvfile)
+
+        blockSize = self.get_BlockSize(self.params.N, csv_file)
+
+        # for node in xrange(self.params.N):
+        extra_args = extra.copy()
+        for node in xrange(self.params.N):
+            jobname = "gcsvBulk-%s" % node
+
+            extra_args = extra.copy()
+
+            extra_args['output_dir'] = self.params.output
+            extra_args['output_dir'] = extra_args['output_dir'].replace('NAME', 'run_%s' % jobname)
+            extra_args['output_dir'] = extra_args['output_dir'].replace('SESSION', 'run_%s' % jobname)
```

Comment:
Why do you want to substitute `SESSION`, `TIME`, and `DATE` with the job *name* ?